From: SeongJae Park <sj@kernel.org>
Date: Mon, 27 Oct 2025 21:36:23 -0700
Subject: [PATCH] ==== kunit memory bugs fix ====

mm/damon/tests: fix memory bugs in kunit tests

DAMON kunit tests were initially written assuming those will be run on
environments that are well controlled and therefore tolerant to
transient test failures and bugs in the test code itself.  The user-mode
linux based manual run of the tests is one example of such an
environment.  And the test code was written for adding more test
coverage as fast as possible, over making those safe and reliable.

As a result, the tests resulted in having a number of bugs including
real memory leaks, theoretical unhandled memory allocation failures, and
unused memory allocations.  The allocation failures that are not handled
well are unlikely in the real world, since those allocations are too
small to fail.  But in theory, it can happen and cause inappropriate
memory access.

It is arguable if bugs in test code can really harm users.  But, anyway
bugs are bugs that need to be fixed.  Fix the bugs one by one.  Also Cc
stable@ for the fixes of memory leak and unhandled memory allocation
failures.  The unused memory allocations are only a matter of memory
efficiency, so not Cc-ing stable@.

The first patch fixes memory leaks in the test code for the DAMON core
layer.

Following fifteen, three, and one patches respectively fix unhandled
memory allocation failures in the test code for DAMON core layer,
virtual address space DAMON operation set, and DAMON sysfs interface,
one by one per test function.

Final two patches remove memory allocations that are correctly
deallocated at the end, but not really being used by any code.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_dd80b1jr | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_dd80b1jr

diff --git a/hkml_cv_bogus/hkml_cv_bogus_dd80b1jr b/hkml_cv_bogus/hkml_cv_bogus_dd80b1jr
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.47.3

